### PR TITLE
OCPBUGS-30304: cert rotation: Bump library-go to latest master

### PR DESCRIPTION
**Pull request by vrutkovs**
Cherrypick of https://github.com/openshift/cluster-kube-apiserver-operator/pull/1651 on 4.15

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,27 +9,27 @@ require (
...
+	github.com/google/go-cmp v0.6.0
 	github.com/imdario/mergo v0.3.8
 	github.com/miekg/dns v1.1.25
-	github.com/openshift/api v0.0.0-20231129134630-a782d1c1541c
+	github.com/openshift/api v0.0.0-20231218131639-7a5aa77cc72d
 	github.com/openshift/build-machinery-go v0.0.0-20230228230858-4cd708338479
-	github.com/openshift/client-go v0.0.0-20230926161409-848405da69e1
-	github.com/openshift/library-go v0.0.0-20231130112926-2285c05489d7
+	github.com/openshift/client-go v0.0.0-20231218140158-47f6d749b9d9
+	github.com/openshift/library-go v0.0.0-20240305144041-18ee8279b4e3
```
Comment (deads2k) [inline go.mod:18]: /hold

this is bringing 4.16 library-go into 4.15 kube-apiserver-operator. Unless or until we need to update client-go from an unsupported level, this should remain in sync

Comment (vrutkovs) [reply go.mod:18]: My fault, https://github.com/openshift/library-go/pull/1688 is 4.15 cherrypick for library-go

Comment (vrutkovs) [reply go.mod]: Updated

File: go.mod [hunk 2]
```diff
@@ -9,27 +9,27 @@ require (
...
-	k8s.io/klog/v2 v2.100.1
+	github.com/stretchr/testify v1.8.4
+	go.etcd.io/etcd/client/v3 v3.5.10
+	golang.org/x/sys v0.13.0
+	k8s.io/api v0.29.0
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.0
+	k8s.io/apiserver v0.29.0
+	k8s.io/client-go v0.29.0
+	k8s.io/component-base v0.29.0
```
[the remark this answers is not in the document]
Comment (benluddy) [reply go.mod:31]: /hold

The bump commit looks like the result of bumping library-go to the current release-4.15 after bumping to 4.16 and tidying (all of the other dependency changes that came along with 4.16 were preserved). I see a one-line go.mod diff if I revert the other changes from accidental 4.16 bump first: https://github.com/benluddy/cluster-kube-apiserver-operator/commit/d1d767fa8dc9d59febce0d7da5acb6420976a8aa#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

[the remark this answers is not in the document]
Comment (vrutkovs) [reply go.mod]: Updated
/hold cancel

**Pull request merged**